common/utils.h

```diff
@@ -20,6 +20,10 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef __cplusplus
```

Comment:
Now, I have to merge from master, right ? I did it, but I see a special commit in the log about that which breaks our rule about "clean commit history". Will we see this commit in the master after I merge this branch ? Is it correct way to do that ?